crates/common/src/slot_identifier.rs

```diff
@@ -0,0 +1,685 @@
+                    && let Some(slot_info) = self.handle_array_slot(
+                        storage,
+                        storage_type,
+                        slot_u256,
+                        array_start_slot,
+                        &slot_str,
+                    )
+                {
+                    return Some(slot_info);
+                }
+
+                // If type parsing fails and the label is a struct
+                if storage_type.label.starts_with("struct ")
+                    && let Some(slot_info) =
+                        self.handle_struct(storage, storage_type, slot_u256, &slot_str)
+                {
+                    return Some(slot_info);
+                }
+            } else if storage_type.encoding == "mapping"
+                && let Some(mapping_slots) = mapping_slots
+                && let Some(slot_info) =
+                    self.handle_mapping(storage, storage_type, slot, &slot_str, mapping_slots)
+            {
+                return Some(slot_info);
+            }
+        }
+
+        None
+    }
+
+    fn handle_array_slot(
+        &self,
+        storage: &Storage,
+        storage_type: &StorageType,
+        slot: U256,
+        array_start_slot: U256,
+        slot_str: &str,
+    ) -> Option<SlotInfo> {
+        // Check if slot is within array bounds
+        let total_bytes = storage_type.number_of_bytes.parse::<u64>().ok()?;
+        let total_slots = total_bytes.div_ceil(32);
+
+        if slot >= array_start_slot && slot < array_start_slot + U256::from(total_slots) {
+            let parsed_type = DynSolType::parse(&storage_type.label).ok()?;
+            let index = (slot - array_start_slot).to::<u64>();
+            // Format the array element label based on array dimensions
+            let label = match &parsed_type {
+                DynSolType::FixedArray(inner, _) => {
+                    if let DynSolType::FixedArray(_, inner_size) = inner.as_ref() {
+                        // 2D array: calculate row and column
+                        let row = index / (*inner_size as u64);
+                        let col = index % (*inner_size as u64);
+                        format!("{}[{row}][{col}]", storage.label)
+                    } else {
+                        // 1D array
+                        format!("{}[{index}]", storage.label)
+                    }
+                }
+                _ => storage.label.clone(),
+            };
+
+            return Some(SlotInfo {
+                label,
+                slot_type: StorageTypeInfo {
+                    label: storage_type.label.clone(),
+                    dyn_sol_type: parsed_type,
+                },
+                offset: 0,
+                slot: slot_str.to_string(),
+                members: None,
+                decoded: None,
+                keys: None,
+            });
+        }
+
+        None
+    }
+
+    fn handle_struct(
```

Comment:
same as above, let's add a comment to this fn too. Do we need both `handle_struct` and `handle_struct_recursive` too or can we get rid of `handle_struct` and rename `handle_struct_recursive` to `handle_struct`?